### PR TITLE
Support method chaining in charts

### PR DESCRIPTION
**Pull request by kinegratii**
This PR provides a simple form to show chart building process with chain call.

```
config() => add() => render()
```

This is **syntactic sugar** and it will not apply any change in feature and performance.

Example:

```python
from pyecharts import Bar

CLOTHES = ["衬衫", "羊毛衫", "雪纺衫", "裤子", "高跟鞋", "袜子"]
clothes_v1 = [5, 20, 36, 10, 75, 90]
clothes_v2 = [10, 25, 8, 60, 20, 80]

Bar("柱状图数据堆叠示例").add(
    "商家A", CLOTHES, clothes_v1, is_stack=True
).add(
    "商家B", CLOTHES, clothes_v2, is_stack=True
).render()

```

### CODE REVIEW

**Comment by chenjiandongx** [comment]
@kinegratii 请使用版本为 18.4a4 的 black，
```bash
$ pip install black==18.4a4
$ pip install pink
$ pink */*.py
```

**Comment by kinegratii** [comment]
Yeah, we should make a set of command & script for development.

e.g.
```
requirements_dev.txt
code_format.sh
```

**Comment by chenjiandongx** [comment]
你上面的还是不太对，你的 black 版本对了吗

```python
D:\Python\pyecharts>black --version
black, version 18.4a4
```

File: pyecharts/chart.py [hunk 1]
```diff
@@ -374,7 +374,7 @@ def add(
         """
         `add()` 方法只是用于提供自动参数补全
         """
-        pass
+        return self
```
Comment (chfw) [inline pyecharts/chart.py]: Please update this method as step 1: 

```
def add(*args, **kwargs):
    self.__add(*args, **kwargs)
    return self
```

File: pyecharts/charts/bar.py [hunk 2]
```diff
@@ -15,6 +15,7 @@ def __init__(self, title="", subtitle="", **kwargs):
 
     def add(self, *args, **kwargs):
```
Comment (chfw) [inline pyecharts/charts/bar.py]: As step 2, please delete this add method here and elsewhere in charts folder.

**Comment by kinegratii** [comment]
```
class Chart:
    def add(self, *args, **kwargs):
        self.__add(*args, **kwargs)
```

The solution will lose the kwargs parameters autocomplete.

https://github.com/pyecharts/pyecharts/blob/6a33e726e379e120a7257abadfac87551f612f5a/pyecharts/chart.py#L374-L378

The parent `Chart.add()` is just a declaration and is not covered in code runtime. 

So it is a choice between **Program-friendly(Autocomplete)** and **Code-Readable(OOP)**,I think it is better to keep autocomplete.

**Comment by chenjiandongx** [comment]
@chfw 毫无疑问是 python-friendly。

@kinegratii 对应文档及 changelog 请更新

**Pull request merged**